Detectors/ITSMFT/MFT/tracking/src/IOUtils.cxx

```diff
@@ -44,16 +45,18 @@ int ioutils::loadROFrameData(const o2::itsmft::ROFRecord& rof, ROframe& event, g
   auto first = rof.getFirstEntry();
   auto clusters_in_frame = rof.getROFData(clusters);
   for (auto& c : clusters_in_frame) {
-    int layer = geom->getLayer(c.getSensorID());
+    auto sensorID = c.getSensorID();
+    if (!tracker->getUseMC()) {
+      // convert chip ID HW to SW only if the clusters were produced from decoded digits !
+      sensorID = map.globalChipIDHW2SW(sensorID);
+    }
+    int layer = geom->getLayer(sensorID);
     auto pattID = c.getPatternID();
```

Comment:
just to be sure I understand these line.
When reconstructing with clusters made from digitized MC, they will arrive ordered in c.getSensorID().
For the decoded raw data they come grouped in RU, which in your case does not support sequential SensorID, so you substitute it with sequential  map.globalChipIDHW2SW(sensorID), right? But then you use this SensorID to access the sensor matrix https://github.com/AliceO2Group/AliceO2/blob/bb3b6e391c41c454312040f38d459b9ab6dff501/Detectors/ITSMFT/MFT/tracking/src/IOUtils.cxx#L71, how this can work correctly in both cases?

Isn't it easier to simply go over the decoded digits or clusters ROF by ROF and order them SensorID?

Reply:
Reordering decoded digits per ROF is also a solution, but I understood that adding CPU for this correction is to be avoided. 
To answer your question: for MC, c.getSensorID() is not ordered and it corresponds directly to the argument for geom->getLayer(sensorID) or geom->getMatrixL2G(sensorID) without any transformation. When decoding from raw data, c.getSensorID() is reshuffled and before calling geom->getLayer(sensorID) or geom->getMatrixL2G(sensorID) one has to convert it back to the unordered value with map.globalChipIDHW2SW(sensorID). The difference between the two cases is detected in the code exclusively by the option --disable-mc given to the workflow. One could introduce an additional option, of course.

Reply:
Hi @shahor02 I see some problems with my testing when I increase statistics, I would like to look more before eventually merging.

Reply:
Hmm, still don't understand: in the hits->digits->clusters the data are ordered in the same SensorID which is assigned in simulation to hits. So, these data are ordered as needed.
For the raw decoded data, I think the cleanest would be to implement `RawPixelDecoder<Mapping>::getNextChipData` specialization for MFT: there, immediately after the  `decodeNextTrigger()` you can build e.g. a `std::vector<o2::itsmft::ChipPixelData*> pointing on the elements of  `RUDecodeData::chipsData` in SensorID increasing order.
Then, instead of looping over RUs and their chips, you can simply loop over this vector of pointer.

Making the ordering and program flow dependent on the disableMC flag is error prone, actually, we quite often run MC data reconstruction with disableMC flag, e.g. to be able to process large volumes of data (which would not fit to the memory if MC labels are added).

Reply:
Ok, good argument. I will have a look at your proposal, initially, I did not want to modify common ITSMFT code.